core/src/main/java/org/apache/calcite/avatica/AvaticaResultSet.java

```diff
@@ -398,15 +403,15 @@ public BigDecimal getBigDecimal(String columnLabel) throws SQLException {
   }
 
   public boolean isBeforeFirst() throws SQLException {
-    return row < 0;
```

Comment:
Couldn't this just be `row < 1` to avoid introducing the `beforeFirst` variable?

Reply:
Ah makes sense. Thanks for the clarification.